cluster/sbin/snappy-nodes.sh

```diff
@@ -338,12 +338,41 @@ function getNumLeadsOnHost() {
   echo $numLeadsOnHost
 }
 
+function scpConf() {
+ var=$(ip addr | grep 'state UP' -A2 | tail -n1 | awk '{print $2}' | cut -f1  -d'/')
+ if [[ "$host" != "$var" && "$host" != "localhost" ]] ; then
+  	#loop to get the all the files avaliable in Conf directory
+	for entry in "${SPARK_CONF_DIR}"/*
+	do
+	  if [ -f "$entry" ];then
+	    #${file%.*} to get the filename without the extension and ${file##*.} to get the extension alone
+	    fileName=$(basename $entry)
+	    #echo "fileName:  $fileName"
+   	    template=".template"
+	    #check the extension, interested in files those doesn't have template extension
+	    if [[ ! "$fileName" = @(*.template) ]]; then	       	
+		if ! ssh $host "test -e $entry"; then #"File does not exist." 		  
+		  scp -i ~/.ssh/id_rsa.pub ${SPARK_CONF_DIR}/$fileName  $host:${SPARK_CONF_DIR}
```

Comment:
passwordless ssh is assumed to exists for the user. Direct scp should work without the '-i' option.

Reply:
Removed -i option